zerver/models.py

```diff
@@ -1094,6 +1096,23 @@ def is_claimed(self):
     def get_url(self):
         return "/user_uploads/%s" % (self.path_id)
 
+
+def validate_attachment_request(user_profile, path_id):
+    try:
+        attachment = Attachment.objects.get(path_id=path_id)
+        messages = attachment.messages.all()
```

Comment:
I think this is probably good enough from a queries perspective; I'd expect this version ot be fairly efficient.